config_manager.go

```diff
@@ -50,105 +49,28 @@ type ConfigManager struct {
 		UseTCP          bool
 		UseTLS          bool
 		NoDaemonize     bool
-		Severity        string
-		Facility        string
+		Severity        syslog.Priority
+		Facility        syslog.Priority
 		Poll            bool
 	}
 }
 
-type RefreshInterval struct {
-	Duration time.Duration
-}
-
-func (r *RefreshInterval) String() string {
-	return fmt.Sprint(*r)
-}
-
-func (r *RefreshInterval) Set(value string) error {
-	d, err := time.ParseDuration(value)
-
-	if err != nil {
-		return err
+func NewConfigManager() (*ConfigManager, error) {
+	cm := &ConfigManager{
+		Config: ConfigFile{
+			ExcludeFiles:    RegexCollection{},
+			ExcludePatterns: RegexCollection{},
+		},
 	}
-
-	if d < MinimumRefreshInterval {
-		return fmt.Errorf("refresh interval must be greater than %s", MinimumRefreshInterval)
-	}
-	r.Duration = d
-	return nil
-}
-
-func (r *RefreshInterval) SetYAML(tag string, value interface{}) bool {
-	err := r.Set(value.(string))
-	if err != nil {
-		return false
-	}
-	return true
-}
-
-type RegexCollection []*regexp.Regexp
-
-func (r *RegexCollection) Set(value string) error {
-	exp, err := regexp.Compile(value)
-	if err != nil {
-		return err
-	}
-	*r = append(*r, exp)
-	return nil
-}
-
-func (r *RegexCollection) String() string {
-	return fmt.Sprint(*r)
-}
-
-func (r *RegexCollection) SetYAML(tag string, value interface{}) bool {
-	items, ok := value.([]interface{})
-
-	if !ok {
-		return false
-	}
-
-	for _, item := range items {
-		s, ok := item.(string)
-
-		if !ok {
-			return false
-		}
-
-		err := r.Set(s)
-		if err != nil {
-			panic(fmt.Sprintf("Failed to compile regex expression \"%s\"", s))
-		}
-	}
-
-	return true
-}
-
-func NewConfigManager() ConfigManager {
-	cm := ConfigManager{}
-	err := cm.Initialize()
-
-	if err != nil {
-		log.Criticalf("Failed to configure the application: %s", err)
-		os.Exit(1)
-	}
-
-	return cm
-}
-
-func (cm *ConfigManager) Initialize() error {
-	cm.Config.ExcludeFiles = &RegexCollection{}
-	cm.Config.ExcludePatterns = &RegexCollection{}
 	cm.parseFlags()
-
-	err := cm.readConfig()
-	if err != nil {
-		return err
+	if err := cm.readConfig(); err != nil {
+		log.Errorf("Error reading config file: %v", err)
```

Comment:
Do we need to log anything here if we're logging [here](https://github.com/papertrail/remote_syslog2/blob/validate-config/remote_syslog.go#L118)?